openshift/machine-config-operator
# OCPBUGS-3909: Don't validate contents and mode for masked units

**- What I did**

Fixes OCPBUGS-3909. The key here is to ignore the unit contents and file mode when it is masked. I also did a bit of refactoring to reuse the writeFiles and writeSystemdUnits code in the MCD to make this easier to reproduce.

**- How to verify it**

The TestConfigDriftMonitor includes a new test fixture that will cause the test to fail without the implemented fix. There is also a new e2e test case for the Config Drift e2e Test Suite. So one can run either the unit test suite or the e2e suite (or both!). Additionally, to verify my other refactoring changes, the e2e-gcp-op test target should provide adequate coverage for that code.

**- Description for the changelog**
Ignore file content and mode for masked systemd units (fixes OCPBUGS-3909)
